ext/pdo_sqlite/pdo_sqlite_class.c

```diff
@@ -0,0 +1,657 @@
+		stream->eof = 1;
+		sqlite3_stream->position = sqlite3_stream->size;
+	}
+	else {
+		sqlite3_stream->position += count;
+	}
+
+	return count;
+}
+
+static ssize_t php_pdosqlite3_stream_read(php_stream *stream, char *buf, size_t count)
+{
+	php_stream_pdosqlite3_data *sqlite3_stream = (php_stream_pdosqlite3_data *) stream->abstract;
+
+	if (sqlite3_stream->position + count >= sqlite3_stream->size) {
+		count = sqlite3_stream->size - sqlite3_stream->position;
+		stream->eof = 1;
+	}
+	if (count) {
+		if (sqlite3_blob_read(sqlite3_stream->blob, buf, count, sqlite3_stream->position) != SQLITE_OK) {
+			return -1;
+		}
+		sqlite3_stream->position += count;
+	}
+	return count;
+}
+
+static int php_pdosqlite3_stream_close(php_stream *stream, int close_handle)
+{
+	php_stream_pdosqlite3_data *sqlite3_stream = (php_stream_pdosqlite3_data *) stream->abstract;
+
+	if (sqlite3_blob_close(sqlite3_stream->blob) != SQLITE_OK) {
+		/* Error occurred, but it still closed */
+	}
+
+	efree(sqlite3_stream);
+
+	return 0;
+}
+
+static int php_pdosqlite3_stream_flush(php_stream *stream)
+{
+	/* do nothing */
+	return 0;
+}
+
+/* {{{ */
+static int php_pdosqlite3_stream_seek(php_stream *stream, zend_off_t offset, int whence, zend_off_t *newoffs)
+{
+	php_stream_pdosqlite3_data *sqlite3_stream = (php_stream_pdosqlite3_data *) stream->abstract;
+
+	switch(whence) {
+		case SEEK_CUR:
+			if (offset < 0) {
+				if (sqlite3_stream->position < (size_t)(-offset)) {
+					sqlite3_stream->position = 0;
+					*newoffs = -1;
+					return -1;
+				} else {
+					sqlite3_stream->position = sqlite3_stream->position + offset;
+					*newoffs = sqlite3_stream->position;
+					stream->eof = 0;
+					return 0;
+				}
+			} else {
+				if (sqlite3_stream->position + (size_t)(offset) > sqlite3_stream->size) {
+					sqlite3_stream->position = sqlite3_stream->size;
+					*newoffs = -1;
+					return -1;
+				} else {
+					sqlite3_stream->position = sqlite3_stream->position + offset;
+					*newoffs = sqlite3_stream->position;
+					stream->eof = 0;
+					return 0;
+				}
+			}
+		case SEEK_SET:
+			if (sqlite3_stream->size < (size_t)(offset)) {
+				sqlite3_stream->position = sqlite3_stream->size;
+				*newoffs = -1;
+				return -1;
+			} else {
+				sqlite3_stream->position = offset;
+				*newoffs = sqlite3_stream->position;
+				stream->eof = 0;
+				return 0;
+			}
+		case SEEK_END:
+			if (offset > 0) {
+				sqlite3_stream->position = sqlite3_stream->size;
+				*newoffs = -1;
+				return -1;
+			} else if (sqlite3_stream->size < (size_t)(-offset)) {
+				sqlite3_stream->position = 0;
+				*newoffs = -1;
+				return -1;
+			} else {
+				sqlite3_stream->position = sqlite3_stream->size + offset;
+				*newoffs = sqlite3_stream->position;
+				stream->eof = 0;
+				return 0;
+			}
+		default:
+			*newoffs = sqlite3_stream->position;
+			return -1;
+	}
+}
+/* }}} */
+
+
+static int php_pdosqlite3_stream_cast(php_stream *stream, int castas, void **ret)
+{
+	return FAILURE;
+}
```

Comment:
Currently this will cause an issue if using the object in an ``if ($object)`` condition. This is however something that should be fixed at the engine level.

Reply:
In that case it's also a problem in the sqlite3 extension, which is where I copied the code from.

Reply:
I'll keep a memo for that :)